catalog/app/components/TalkToUs/TalkToUs.js

```diff
@@ -36,42 +36,50 @@ function getCalendlyPromise() {
   return window[CALENDLY_PROMISE]
 }
 
-// const getCalendlyEvent = (e) =>
-//   e.data.event &&
-//   e.data.event.startsWith('calendly.') &&
-//   e.data.event.substring('calendly.'.length)
+function useSingletonListener() {
+  const ref = React.useRef(null)
+  return (eventName, callback) => {
```

Comment:
i'd add smth like `if (ref.current === callback) return` to avoid unnecessary re-subscription of the same listener

Reply:
return value should be memoized via `useCallback` or similar, so that it can be used as a dependency for other memoized functions/values